app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.spouse = None
+        Person.people[name] = self
+
+    def set_spouse(self, spouse_name: str) -> None:
+        spouse = Person.people.get(spouse_name)
+        if spouse:
+            self.spouse = spouse
+            spouse.spouse = self
+            if "wife" in vars(self).keys():
+                setattr(self, "husband", spouse)
+                setattr(spouse, "wife", self)
+            else:
+                setattr(self, "wife", spouse)
+                setattr(spouse, "husband", self)
```

Comment:
The `set_spouse` method could be simplified. Instead of using `vars(self).keys()`, you can directly check if the `self` object has the attribute `wife` using `hasattr(self, 'wife')`. Also, it is not clear why you are setting the attributes `wife` and `husband` dynamically. If these are expected attributes of a `Person`, it would be better to declare them in the `__init__` method.